test/test__base.py

```diff
@@ -7,7 +7,9 @@
 
 def test_check_deterministic(non_deterministic_case):
     """Test that non-deterministic behavior is recognized."""
-    model_func, loss_func, params, data = non_deterministic_case
+    model_func, loss_func, params, data, batch_size_fn = non_deterministic_case
 
     with raises(RuntimeError):
-        _LinearOperator(model_func, loss_func, params, data, check_deterministic=True)
+        _LinearOperator(
+            model_func, loss_func, params, data, batch_size_fn, check_deterministic=True
```

Comment:
use `batch_size_fn=batch_size_fn` to make sure this does not break if we introduce more keyword arguments.